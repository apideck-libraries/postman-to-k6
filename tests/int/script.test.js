/* eslint-disable space-before-function-paren */
import convertFile from '../../lib/convert/file';

test('pre request', async () => {
  const [main] = await convertFile('test/material/2/pre-request.json');
  expect(main).toMatchSnapshot();
});

test('pre collection', async () => {
  const [main] = await convertFile('test/material/2/pre-collection.json');
  expect(main).toMatchSnapshot();
});

test('pre folder', async () => {
  const [main] = await convertFile('test/material/2/pre-folder.json');
  expect(main).toMatchSnapshot();
});

test('pre nested', async () => {
  const [main] = await convertFile('test/material/2/pre-nested.json');
  expect(main).toMatchSnapshot();
});

test('pre request --skip-pre', async () => {
  const options = {
    skip: {
      pre: true,
    },
  };
  const [main] = await convertFile('test/material/2/pre-request.json', options);
  expect(main).toMatchSnapshot();
});

test('pre collection --skip-pre', async () => {
  const options = {
    skip: {
      pre: true,
    },
  };
  const [main] = await convertFile(
    'test/material/2/pre-collection.json',
    options
  );
  expect(main).toMatchSnapshot();
});

test('pre folder --skip-pre', async () => {
  const options = {
    skip: {
      pre: true,
    },
  };
  const [main] = await convertFile('test/material/2/pre-folder.json', options);
  expect(main).toMatchSnapshot();
});

test('pre nested --skip-pre', async () => {
  const options = {
    skip: {
      pre: true,
    },
  };
  const [main] = await convertFile('test/material/2/pre-nested.json', options);
  expect(main).toMatchSnapshot();
});

test('post request', async () => {
  const [main] = await convertFile('test/material/2/post-request.json');
  expect(main).toMatchSnapshot();
});

test('post collection', async () => {
  const [main] = await convertFile('test/material/2/post-collection.json');
  expect(main).toMatchSnapshot();
});

test('post folder', async () => {
  const [main] = await convertFile('test/material/2/post-folder.json');
  expect(main).toMatchSnapshot();
});

test('post nested', async () => {
  const [main] = await convertFile('test/material/2/post-nested.json');
  expect(main).toMatchSnapshot();
});

test('post request --skip-pre', async () => {
  const options = {
    skip: {
      post: true,
    },
  };
  const [main] = await convertFile(
    'test/material/2/post-request.json',
    options
  );
  expect(main).toMatchSnapshot();
});

test('post collection --skip-pre', async () => {
  const options = {
    skip: {
      post: true,
    },
  };
  const [main] = await convertFile(
    'test/material/2/post-collection.json',
    options
  );
  expect(main).toMatchSnapshot();
});

test('post folder --skip-pre', async () => {
  const options = {
    skip: {
      post: true,
    },
  };
  const [main] = await convertFile('test/material/2/post-folder.json', options);
  expect(main).toMatchSnapshot();
});

test('post nested --skip-pre', async () => {
  const options = {
    skip: {
      post: true,
    },
  };
  const [main] = await convertFile('test/material/2/post-nested.json', options);
  expect(main).toMatchSnapshot();
});
