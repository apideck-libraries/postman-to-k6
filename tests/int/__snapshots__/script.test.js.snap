// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post collection --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`post collection 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Post = Symbol.for("post");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Post].push(() => {
    pm.variables.set("test", "a");
    pm.variables.set("test", "b");
    pm.variables.set("test", "c");
  });

  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });

  postman[Post].pop();
}
"
`;

exports[`post folder --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/"
    });
  });
}
"
`;

exports[`post folder 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Post = Symbol.for("post");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Post].push(() => {
      pm.variables.set("test", "a");
      pm.variables.set("test", "b");
      pm.variables.set("test", "c");
    });

    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/"
    });

    postman[Post].pop();
  });
}
"
`;

exports[`post nested --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    group("TestFolder2", function() {
      group("TestFolder3", function() {
        postman[Request]({
          name: "TestRequest",
          method: "GET",
          address: "http://example.com/"
        });
      });
    });
  });
}
"
`;

exports[`post nested 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Post = Symbol.for("post");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Post].push(() => {
      pm.variables.set("test", "a");
    });

    group("TestFolder2", function() {
      postman[Post].push(() => {
        pm.variables.set("test", "b");
      });

      group("TestFolder3", function() {
        postman[Post].push(() => {
          pm.variables.set("test", "c");
        });

        postman[Request]({
          name: "TestRequest",
          method: "GET",
          address: "http://example.com/"
        });

        postman[Post].pop();
      });

      postman[Post].pop();
    });

    postman[Post].pop();
  });
}
"
`;

exports[`post request --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`post request 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    post(response) {
      pm.variables.set("test", "a");
      pm.variables.set("test", "b");
      pm.variables.set("test", "c");
    }
  });
}
"
`;

exports[`pre collection --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`pre collection 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Pre = Symbol.for("pre");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Pre].push(() => {
    pm.variables.set("test", "a");
    pm.variables.set("test2", "b");
    pm.variables.set("test3", "c");
  });

  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });

  postman[Pre].pop();
}
"
`;

exports[`pre folder --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/"
    });
  });
}
"
`;

exports[`pre folder 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Pre = Symbol.for("pre");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Pre].push(() => {
      pm.variables.set("test", "a");
      pm.variables.set("test2", "b");
      pm.variables.set("test3", "c");
    });

    postman[Request]({
      name: "TestRequest",
      method: "GET",
      address: "http://example.com/"
    });

    postman[Pre].pop();
  });
}
"
`;

exports[`pre nested --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    group("TestFolder2", function() {
      group("TestFolder3", function() {
        postman[Request]({
          name: "TestRequest",
          method: "GET",
          address: "http://example.com/"
        });
      });
    });
  });
}
"
`;

exports[`pre nested 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";
import { group } from "k6";

export let options = { maxRedirects: 4 };

const Pre = Symbol.for("pre");
const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  group("TestFolder", function() {
    postman[Pre].push(() => {
      pm.variables.set("test", "a");
    });

    group("TestFolder2", function() {
      postman[Pre].push(() => {
        pm.variables.set("test2", "b");
      });

      group("TestFolder3", function() {
        postman[Pre].push(() => {
          pm.variables.set("test3", "c");
        });

        postman[Request]({
          name: "TestRequest",
          method: "GET",
          address: "http://example.com/"
        });

        postman[Pre].pop();
      });

      postman[Pre].pop();
    });

    postman[Pre].pop();
  });
}
"
`;

exports[`pre request --skip-pre 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/"
  });
}
"
`;

exports[`pre request 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "TestRequest",
    method: "GET",
    address: "http://example.com/",
    pre() {
      pm.variables.set("test", "a");
      pm.variables.set("test2", "b");
      pm.variables.set("test3", "c");
    }
  });
}
"
`;
